Engine/Results/BaseResultsHandler.cs

```diff
@@ -441,52 +441,42 @@ protected virtual decimal GetPortfolioValue()
         /// </summary>
         /// <remarks>Useful so that live trading implementation can freeze the returned value if there is no user exchange open
         /// so we ignore extended market hours updates</remarks>
-        protected virtual decimal GetBenchmarkValue()
+        protected virtual decimal GetBenchmarkValue(DateTime time)
         {
-            return Algorithm.Benchmark.Evaluate(PreviousUtcSampleTime).SmartRounding();
+            return Algorithm.Benchmark.Evaluate(time).SmartRounding();
         }
 
         /// <summary>
         /// Samples portfolio equity, benchmark, and daily performance
+        /// Called by scheduled event every night at midnight algorithm time
         /// </summary>
         /// <param name="time">Current UTC time in the AlgorithmManager loop</param>
         /// <param name="force">Force sampling of equity, benchmark, and performance to be </param>
+        /// <remarks>Force is no longer used</remarks>
         public virtual void Sample(DateTime time, bool force = false)
         {
-            var dayChanged = PreviousUtcSampleTime.Date != time.Date;
-
-            if (dayChanged || force)
+            var currentPortfolioValue = GetPortfolioValue();
+            var portfolioPerformance = DailyPortfolioValue == 0 ? 0 : Math.Round((currentPortfolioValue - DailyPortfolioValue) * 100 / DailyPortfolioValue, 10);
+
+            // Update our max portfolio value
+            CumulativeMaxPortfolioValue = Math.Max(currentPortfolioValue, CumulativeMaxPortfolioValue);
+
+            // Sample all our default charts
+            SampleEquity(time, currentPortfolioValue);
+            SampleBenchmark(time, GetBenchmarkValue(time));
+            SamplePerformance(time, portfolioPerformance);
+            SampleDrawdown(time, currentPortfolioValue);
+            SampleSalesVolume(time);
+            SampleExposure(time, currentPortfolioValue);
+            SampleCapacity(time);
+
+            // Update daily portfolio value only once a day on the first point of the day
+            if (PreviousUtcSampleTime.Date != time.Date)
```

Comment:
Since we are using the scheduled event, is this `if` still required? 🤔 We might be able to just drop `PreviousUtcSampleTime` entirely?